README.md
[![](https://ptkdev.it/img/bot/social-manager-tools.png)](https://socialmanagertools.ptkdev.io)

# Social Manager Tools

[![](https://img.shields.io/badge/license-AGLPv3-brightgreen.svg)](#) [![](https://img.shields.io/badge/version-v0.6.2-lightgrey.svg)](https://github.com/social-manager-tools/social-manager-tools/releases) [![](https://img.shields.io/badge/chat%20on-Slack-orange.svg)](https://slack.ptkdev.io) [![](https://img.shields.io/badge/blog-medium-2AE176.svg)](http://blog.ptkdev.io) [![](https://img.shields.io/badge/twitter-ptkdevio-2AA3EF.svg)](https://twitter.com/ptkdevio) [![](https://img.shields.io/badge/help-support@ptkdev.io-fbbc05.svg)](mailto:support@ptkdev.io)

[![](https://img.shields.io/badge/powered%20by-instagrambotlib-895a4d.svg)](https://github.com/social-manager-tools/instagram-bot-lib) [![](https://img.shields.io/badge/powered%20by-twitterbotlib-21B7F4.svg)](https://github.com/social-manager-tools/twitter-bot-lib) [![](https://img.shields.io/badge/powered%20by-facebookbotlib-3b5998.svg)](https://github.com/social-manager-tools/facebook-bot-lib)

[![](https://img.shields.io/badge/donate-patreon-F87668.svg)](http://patreon.ptkdev.io) [![](https://img.shields.io/badge/donate-paypal-46AFE0.svg)](http://paypal.ptkdev.io) [![](https://img.shields.io/badge/buy%20me-coffee-4B788C.svg)](http://coffee.ptkdev.io)

[![](https://ptkdev.it/img/bot/social-manager-tools.gif)](https://instagram.bot.ptkdev.io)

## What does it do
This app helps you increase the engagement of your social profile (instagram, twitter, facebook) through different social algorithms. Increase the likes on your photos and followers.

## Features
* [✓] Easy to use
* [✓] GUI for [InstagramBot.js](https://github.com/social-manager-tools/instagram-bot.js) 
* [✓] GUI for [TwitterBot.js](https://github.com/social-manager-tools/twitter-bot.js) 
* [✘] GUI for [FacebookPageBot.js](https://github.com/social-manager-tools/facebookpage-bot.js) 
* [✘] GUI for [Telegram Bot for Wordpress](https://github.com/social-manager-tools/wordpress-telegram-bot.js) 
* [✘] GUI for [Telegram Bot for Medium](https://github.com/social-manager-tools/medium-telegram-bot.js) 

## Fast Setup
1. Download [stable version](https://socialmanagertools.ptkdev.io) and install it.

## Developer Setup
1. Download [stable version](https://github.com/social-manager-tools/social-manager-tools/releases) and extract it.
2. Download [Node.js](https://nodejs.org/it/) >= 7.6 and install it.
3. Run `npm install electron electron-builder -g`
4. Run `npm install` in `social-manager-tools` folder.
5. If you received error with sqllite3 problem, run `npm run build-fix`
6. Start the ui bot via `npm run start` or build via `npm run dist`
7. If work add star :star: at this project :heart:
8. If you want help me: **[donate on paypal](http://paypal.ptkdev.io)** or become a **[backer on patreon](http://patreon.ptkdev.io)**.

## Sorry for snake_case
I love :snake: snake_case syntax sorry for this :sob: don't hate me.

[![](https://socialmanagertools.ptkdev.io/img/socialmanagertools_logo.png)](https://github.com/social-manager-tools)

# Social Manager Tools

[Social Manager Tools GUI](https://github.com/social-manager-tools/social-manager-tools)  
[InstagramBot.js](https://github.com/social-manager-tools/instagram-bot.js) ([LIB](https://github.com/social-manager-tools/instagram-bot-lib))  
[TwitterBot.js](https://github.com/social-manager-tools/twitter-bot.js) ([LIB](https://github.com/social-manager-tools/twitter-bot-lib))  
[FacebookBot.js](https://github.com/social-manager-tools/facebook-bot.js) ([LIB](https://github.com/social-manager-tools/facebook-bot-lib))  
[WordpressTelegramBot.js](https://github.com/social-manager-tools/wordpress-telegram-bot.js) ([LIB](https://github.com/social-manager-tools/wordpress-telegram-bot-lib))  
[MediumTelegramBot.js](https://github.com/social-manager-tools/medium-telegram-bot.js) ([LIB](https://github.com/social-manager-tools/medium-telegram-bot-lib))  

# License

GNU AFFERO GENERAL PUBLIC LICENSE

Copyright (c) 2018 Patryk Rzucidło (PTKDev)


CHANGELOG.md
# v0.6.2 (November 28, 2018)
* Update: [instagrambotlib v0.9.10](https://github.com/social-manager-tools/instagram-bot-lib/blob/master/CHANGELOG.md)


# v0.6.1 (October 16, 2018)
* Fix: Wrong instructions on where to find Post you've liked #41
* Update: [instagrambotlib v0.9.9](https://github.com/social-manager-tools/instagram-bot-lib/blob/master/CHANGELOG.md)


# v0.6.0 (October 14, 2018)
* Fix: 2FA Work
* Fix: bot remove space in comment_mode (instagrambotlib)
* Update: [instagrambotlib v0.9.8](https://github.com/social-manager-tools/instagram-bot-lib/blob/master/CHANGELOG.md)


# v0.5.6 (October 05, 2018)
* Update: [instagrambotlib v0.9.6](https://github.com/social-manager-tools/instagram-bot-lib/blob/master/CHANGELOG.md)


# v0.5.5 (October 02, 2018)
* Update: [instagrambotlib v0.9.5](https://github.com/social-manager-tools/instagram-bot-lib/blob/master/CHANGELOG.md)
* Update: [twitterbotlib v0.2.5](https://github.com/social-manager-tools/twitter-bot-lib/blob/master/CHANGELOG.md)


# v0.5.4 (September 26, 2018)
* Update: [instagrambotlib v0.9.4](https://github.com/social-manager-tools/instagram-bot-lib/blob/master/CHANGELOG.md)
* Update: [twitterbotlib v0.2.4](https://github.com/social-manager-tools/twitter-bot-lib/blob/master/CHANGELOG.md)


# v0.5.3 (September 22, 2018)
* Update: [instagrambotlib v0.9.3](https://github.com/social-manager-tools/instagram-bot-lib/blob/master/CHANGELOG.md)
* Fix: various misprint and minor bug
* Feature: stop button work (instagrambot)


# v0.5.2 (September 15, 2018)
* Update: [instagrambotlib v0.9.2](https://github.com/social-manager-tools/instagram-bot-lib/blob/master/CHANGELOG.md)
* Update: [twitterbotlib v0.2.3](https://github.com/social-manager-tools/twitter-bot-lib/blob/master/CHANGELOG.md)


# v0.5.1 (September 15, 2018)
* Fix: in fdfmode_classic wrong check and bad alert
* Fix: clean log button after restart bot
* Update: [instagrambotlib v0.9.1](https://github.com/social-manager-tools/instagram-bot-lib/blob/master/CHANGELOG.md)
* Update: [twitterbotlib v0.2.2](https://github.com/social-manager-tools/twitter-bot-lib/blob/master/CHANGELOG.md)


# v0.5.0 (September 12, 2018)
* Feature (instagram): fdfmode_classic
* Feature (instagram): comment_mode
* Feature (instagram): likemode_competitor_users
* Fix: stop bouncing chrome icon on macOS
* Update: [instagrambotlib v0.9.0c](https://github.com/social-manager-tools/instagram-bot-lib/blob/master/CHANGELOG.md)
* Update: [twitterbotlib v0.2.1](https://github.com/social-manager-tools/twitter-bot-lib/blob/master/CHANGELOG.md)


# v0.4.2 (July 31, 2018)
* Feature: icons png to webp
* Fix: multiple popup of update message
* Update: [instagrambotlib v0.8.2](https://github.com/social-manager-tools/instagram-bot-lib/blob/master/CHANGELOG.md)
* Update: [twitterbotlib v0.2.0](https://github.com/social-manager-tools/twitter-bot-lib/blob/master/CHANGELOG.md)
* Coming Soon: Facebook Bot v0.1.0 
* Coming Soon: Telegram for Wordpress v0.1.0 
* Coming Soon: Telegram for Medium v0.1.0 


# v0.4.1 (July 23, 2018)
* Fix: missing build default config (twitterbot)


# v0.4.0 (July 23, 2018)
* Fix: On macOS close button don't close app
* Feature: TwitterBot v0.1.0
* Feature: likemode_realistic (twitterbot)
* Feature: rtmode_realistic (twitterbot)
* Feature: 2FA Support (twitterbot)


# v0.3.1 (July 11, 2018)

* Fix: download url for update app
* Fix: likemode_superlike timeout error
* Fix: Auto-refresh logs popup
* Feature: disable start button if bot is started
* Removed: tray icon on mac xos


# v0.3.0 (July 09, 2018)

* Feature: tray icon 
* Feature: icons on menu near labels
* Feature: Debugmode
* Feature: Error logs file
* Feature: 2FA pin (instagrambot)
* Feature: Set max and min like/day (instagrambot)
* Feature: likemode_classic (instagrambot)
* Feature: likemode_superlike (instagrambot)
* Coming Soon: TwitterBot v0.1.0 


# v0.2.2 (July 03, 2018)

* Update: instagram-bot-lib v0.7.5
* Fix: Update system
* Fix: Autoremove # from hashtag list
* Fix: Size app (~102 -> ~42mb)
* Fix: Random, bot insert more than 900 like/day


# v0.2.1 (June 29, 2018)

* Feature: New UI
* Feature: Multi account
* Update: InstagramBot v0.7.4


# v0.1.0 BETA (March 19, 2018)

* First Release


main.js
// Modules to control application life and create native browser window
const { app, BrowserWindow, Tray, Menu } = require("electron");
const path = require("path");
// Keep a global reference of the window object, if you don't, the window will
// be closed automatically when the JavaScript object is garbage collected.
let main_window;
let app_icon, context_menu;
var sessions = [];

require("electron-context-menu")({
    prepend: (params, BrowserWindow) => [{
        labels: {
            cut: "Configured Cut",
            copy: "Configured Copy",
            paste: "Configured Paste"
        },

        // Only show it when right-clicking images
        visible: params.mediaType === "input"
    }]
});

function create_window() {
    // Create the browser window.
    main_window = new BrowserWindow({
        width: 1024,
        height: 768,
        title: "Social Manager Tools",
        icon: path.join(__dirname, "/src/img/smt_logo.png")
    });

    if (process.platform == "win32") {
        app_icon = new Tray(path.join(__dirname, "/src/img/smt_logo.ico"));
    } else if (process.platform != "darwin") {
        app_icon = new Tray(path.join(__dirname, "/src/img/smt_logo.png"));
    }

    if (process.platform != "darwin") {
        context_menu = Menu.buildFromTemplate([{
            label: "Quit",
            click: function() {
                app.isQuiting = true;
                app.quit();
            }
        }]);
    }

    main_window.loadFile("src/index.html");

    if (process.platform != "darwin") {
        app_icon.on("click", function(event) {
            event.preventDefault();
            if (main_window.isVisible()) {
                main_window.hide();
            } else {
                main_window.show();
            }
        });

        main_window.on("minimize", function(event) {
            event.preventDefault();
            main_window.hide();
        });

        main_window.on("show", function() {
            app_icon.setHighlightMode("always");
        });

        app_icon.setContextMenu(context_menu);
    }

    // Emitted when the window is closed.
    main_window.on("closed", function() {
        // Dereference the window object, usually you would store windows
        // in an array if your app supports multi windows, this is the time
        // when you should delete the corresponding element.
        main_window = null;
    });
}

// This method will be called when Electron has finished
// initialization and is ready to create browser windows.
// Some APIs can only be used after this event occurs.
app.on("ready", create_window);

// Quit when all windows are closed.
app.on("window-all-closed", function() {
    app.quit();
});

app.on("activate", function() {
    // On OS X it's common to re-create a window in the app when the
    // dock icon is clicked and there are no other windows open.
    if (main_window === null) {
        create_window();
    }
});

exports.instagrambot_start = (json, id) => {
    let config = json;
    let Bot = require("instagrambotlib");
    let bot = new Bot(config);
    sessions[id] = bot;
    bot.start();
};

exports.instagrambot_stop = (id) => {
    if (sessions[id] != null) {
        sessions[id].stop();
    }
    sessions[id] = null;
};

exports.twitterbot_start = (json) => {

    let config = json;
    let Bot = require("twitterbotlib");
    let bot = new Bot(config);
    bot.start();
};

configtest.js
module.exports = {
    // InstagramBot.js Configs
    "debug": true,
    "login": true,

    // Instagram Account
    "instagram_username": "ptkdev", // without @
    "instagram_password": "password",
    "instagram_hashtag": ["muraleshunter", "like4like", "follow4follow"], //without #
    "instagram_pin": "sms", //method to receive pin (email or sms)

    // BOT Configs
    // [WORK] likemode_classic          - select random hashtag from config list and like 1 random photo (of last 20) | 400-600 like/day.
    // [WORK] likemode_realistic        - select random hashtag from config list, like fast 10-12 photo and sleep 15-20min. Sleep at night
    // [WORK] likemode_superlike        - select random hashtag from config list and like 3 random photo of same user | 400-600 like/day.
    // [WORK] likemode_competitor_users - select account, select random followers, like 10-12 photo and sleep 15-20min.
    // [WORK] comment_mode              - leave a comment under the photo via hashtag.
    // [TODO] fdfmode_classic           - follow user from random hashtag and defollow after 1h | 300 follow-defollow/day.
    // [TODO] fdfmode_defollowall       - defollow all your following (not defollow users in whitelist) | 30 defollow/hour.
    "bot_mode": "likemode_classic",
    "bot_likeday_min": 400,      // work only in likemode_classic
    "bot_likeday_max": 600,      // work only in likemode_classic. In 2018 limit instagram is descreased to 400-600/day :(
    "bot_sleep_night": true,     // if you need stop bot in night 
    "bot_start_sleep": "7:00",   // sleep from 00:00 to 7:00 am, work in likemode_realistic, comment_mode, likemode_superlike and likemode_competitor_users
    "bot_fastlike_min": "15",    // min minutes of sleep after like 10-12 photo, work in likemode_realistic and likemode_superlike (superlike add 3 like/user, set 50min)
    "bot_fastlike_max": "20",    // max minutes of sleep after like 10-12 photo, work in likemode_realistic and likemode_superlike (superlike add 3 like/user, set 60min)
    "bot_superlike_n": 3,        // superlike number of likes for user
    "bot_followday": 300,        // follow-defollow user/day, work in fdfmode_classic and fdfmode_defollowall
    "bot_userwhitelist": [""],   // usernames without @ | don't defollow this users on follow-defollow mode, work in fdfmode_classic and fdfmode_defollowall
    "comment_mode": {
        "comments": {
            "type": "array",
            "source": [],
        }
    },
    "likemode_competitor_users": {
        "account": "", // without @
    },

    // Puppeteer Configs
    "chrome_headless": false,
    "chrome_options": ["--disable-gpu", "--no-sandbox", "--window-size=1920x1080"],
    "executable_path": "", // example for Mac OS: /Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome

    // LOG
    "log": {
        "drivers": ["console"], // slack or console
        "screenshot": false,  // disable or enable screenshot in logs folder
        "channels": {
            "console": "",
            "slack": {
                "webhook": ""
            }
        }
    }
};


src/js/smt.js
const fs = require("fs");
const remote = require("electron").remote;
const remoteapp = remote.app;
const Jtr = require("json-token-replace");
const jtr = new Jtr();
var list_actived_bot = [];
var logs_interval = null;
$.ajaxSetup({ cache: false });

function update_system() {
    $.get("https://api.ptkdev.io/v1/bot/social-manager-tools/version/?time=" + new Date().getTime(), function(version) {
        let current_version = require("../version").version;
        let download_url = "https://socialmanagertools.ptkdev.io/";
        if (process.platform === "linux") {
            download_url = "https://github.com/social-manager-tools/social-manager-tools/releases/download/" + version + "/Social.Manager.Tools-" + version + ".deb";
        } else if (process.platform === "win32") {
            download_url = "https://github.com/social-manager-tools/social-manager-tools/releases/download/" + version + "/Social.Manager.Tools-" + version + ".zip";
        } else if (process.platform === "darwin") {
            download_url = "https://github.com/social-manager-tools/social-manager-tools/releases/download/" + version + "/Social.Manager.Tools-" + version + ".dmg";
        }

        if (version != current_version) {
            app.dialog.create({ title: "Update available", text: "Your app version is v" + current_version + ", is available v" + version + "<br /><br /><a href='" + download_url + "'>DOWNLOAD</a> | <a href='https://github.com/social-manager-tools/social-manager-tools/blob/master/CHANGELOG.md'>CHANGELOG</a>", buttons: [{ text: "OK" }] }).open();
        }
    });
}

setInterval(function() {
    if(!$(".dialog")[0]) {
        update_system();
    } 
}, (43200 * 1000)); // 12h 
update_system();


src/js/smt-instagram.js
function instagrambot_start(json, id) {
    const main = require("electron").remote.require("./main");
    main.instagrambot_start(json, id);
}

function instagrambot_stop(id) {
    delete list_actived_bot[id];
    const main = require("electron").remote.require("./main");
    main.instagrambot_stop(id);
    app.dialog.create({ title: "Status", text: "Stopped...", buttons: [{ text: "OK" }] }).open();
}

function instagram_get_user_form() {
    let tokens = {};
    tokens.instagram_username = $("#instagram_username").val();
    tokens.instagram_username = tokens.instagram_username.replace(/@/g, "");
    tokens.instagram_password = $("#instagram_password").val();
    tokens.instagram_hashtag = $("#instagram_hashtag").val();
    tokens.bot_mode = $("#bot_mode").val();
    tokens.executable_path = $("#executable_path").val();
    tokens.instagram_hashtag = tokens.instagram_hashtag.replace(/ /g, "");
    tokens.instagram_hashtag = tokens.instagram_hashtag.replace(/#/g, "");
    tokens.instagram_hashtag = tokens.instagram_hashtag.replace(/,/g, "\",\"");

    tokens.bot_likeday_min = parseInt($("#bot_likeday_max").val() - 100);
    tokens.bot_likeday_max = $("#bot_likeday_max").val();

    tokens.bot_followday = $("#bot_followday").val();
    tokens.bot_followrotate = $("#bot_followrotate").val();
    tokens.bot_userwhitelist = $("#bot_userwhitelist").val();
    tokens.bot_userwhitelist = tokens.bot_userwhitelist.replace(/ /g, "");
    tokens.bot_userwhitelist = tokens.bot_userwhitelist.replace(/#/g, "");
    tokens.bot_userwhitelist = tokens.bot_userwhitelist.replace(/,/g, "\",\"");

    if (tokens.bot_mode == "comment_mode") {
        tokens.bot_likeday_min = parseInt($("#bot_commentsday").val() - 50);
        tokens.bot_likeday_max = $("#bot_commentsday").val();
    }

    tokens.bot_comment_list = $("#bot_comment_list").val();
    tokens.bot_comment_list = tokens.bot_comment_list.replace(/,/g, "\",\"");

    tokens.bot_likemode_competitor_users = $("#bot_likemode_competitor_users").val();


    let min = 0;
    do {
        min++;
    } while ((60 / min * 24 * 11) > tokens.bot_likeday_min);
    tokens.bot_fastlike_max = min--;
    min = 0;
    do {
        min++;
    } while ((60 / min * 24 * 11) > tokens.bot_likeday_max);
    tokens.bot_fastlike_min = min--;
    tokens.bot_superlike_n = $("#bot_superlike_n").val();

    if ($("#chrome_headless").val() != "enabled" && $("#chrome_headless").val() != "disabled") {
        tokens.chrome_headless = "disabled";
    } else {
        tokens.chrome_headless = $("#chrome_headless").val();
    }

    if (process.platform === "win32") {
        tokens.config_path = remoteapp.getPath("userData");
        tokens.config_path = tokens.config_path.replace(/\\/g, "/");
    } else {
        tokens.config_path = remoteapp.getPath("userData");
    }

    return tokens;
}

function instagram_load_config() {
    if ($("#executable_path").val() == "") {
        if (process.platform === "linux") {
            $("#executable_path").val("/usr/bin/google-chrome");
        } else if (process.platform === "win32") {
            $("#executable_path").val("C:/Program Files (x86)/Google/Chrome/Application/chrome.exe");
        } else if (process.platform === "darwin") {
            $("#executable_path").val("/Applications/Google Chrome.app/Contents/MacOS/Google Chrome");
        }
    }

    $("#bot_mode").change();

    fs.exists(remoteapp.getPath("userData") + "/config_" + $("#instagram_username").val() + ".json", function(exists) {
        if (exists) {
            let data = fs.readFileSync(remoteapp.getPath("userData") + "/config_" + $("#instagram_username").val() + ".json", "utf8");
            let config = JSON.parse(data.toString());

            $("#instagram_password").val(config.instagram_password);
            $("#bot_mode").val(config.bot_mode);
            $("#bot_likeday_max").val(config.bot_likeday_max);
            $("#bot_superlike_n").val(config.bot_superlike_n);
            $("#executable_path").val(config.executable_path);
            $("#chrome_headless").val(config.chrome_headless);
            $("#instagram_hashtag").val(config.instagram_hashtag.join());

            $("#bot_commentsday").val(config.bot_likeday_max);
            $("#bot_comment_list").val(config.comment_mode.comments.source.join());
            $("#bot_followday").val(config.bot_followday);
            $("#bot_followrotate").val(config.bot_followrotate);
            $("#bot_userwhitelist").val(config.bot_userwhitelist.join());
            $("#bot_likemode_competitor_users").val(config.likemode_competitor_users.account);

            instagram_check_bot_mode();
        }
    });

}

function instagram_check_form() {
    let check_err = 0;

    if ($("#instagram_username").val() == "" && check_err == 0) {
        app.dialog.create({ title: "Warning", text: "Username is empty", buttons: [{ text: "OK" }] }).open();
        check_err++;
    } else if ($("#instagram_password").val() == "" && check_err == 0) {
        app.dialog.create({ title: "Warning", text: "Password is empty", buttons: [{ text: "OK" }] }).open();
        check_err++;
    } else if ($("#instagram_hashtag").val() == "" && $("#bot_mode").val() != "bot_likemode_competitor_users" && check_err == 0) {
        app.dialog.create({ title: "Warning", text: "Hashtag list is empty", buttons: [{ text: "OK" }] }).open();
        check_err++;
    } else if ($("#bot_likeday_max").val() == "" && $("#bot_mode").val() != "fdfmode_classic" && $("#bot_mode").val() != "comment_mode" && check_err == 0) {
        app.dialog.create({ title: "Warning", text: "Max like/day is empty", buttons: [{ text: "OK" }] }).open();
        check_err++;
    } else if ($("#bot_mode").val() == "likemode_superlike" && $("#bot_superlike_n").val() == "" && check_err == 0) {
        app.dialog.create({ title: "Warning", text: "Max Like photo/user is empty", buttons: [{ text: "OK" }] }).open();
        check_err++;
    } else if ($("#bot_commentsday").val() == "" && $("#bot_mode").val() === "comment_mode" && check_err == 0) {
        app.dialog.create({ title: "Warning", text: "Max Comments/day is empty", buttons: [{ text: "OK" }] }).open();
        check_err++;
    } else if ($("#bot_comment_list").val() == "" && $("#bot_mode").val() === "comment_mode" && check_err == 0) {
        app.dialog.create({ title: "Warning", text: "Comments list is empty", buttons: [{ text: "OK" }] }).open();
        check_err++;
    } else if ($("#bot_followday").val() == "" && $("#bot_mode").val() === "fdfmode_classic" && check_err == 0) {
        app.dialog.create({ title: "Warning", text: "Max follow-defollow/day is empty", buttons: [{ text: "OK" }] }).open();
        check_err++;
    } else if ($("#bot_followrotate").val() == "" && $("#bot_mode").val() === "fdfmode_classic" && check_err == 0) {
        app.dialog.create({ title: "Warning", text: "Max follow rotate is empty", buttons: [{ text: "OK" }] }).open();
        check_err++;
    } else if ($("#bot_likemode_competitor_users").val() == "" && $("#bot_mode").val() === "likemode_competitor_users" && check_err == 0) {
        app.dialog.create({ title: "Warning", text: "Insert competitor username", buttons: [{ text: "OK" }] }).open();
        check_err++;
    } else if (check_err == 0) {
        check_err += instagram_check_max_like();
    }

    return check_err;
}

function instagram_save_config(bot) {
    let check_err = instagram_check_form();

    if (list_actived_bot[$("#instagram_username").val()+"_"+$("#bot_mode").val()] == true) {
        app.dialog.create({ title: "Status", text: "@" + $("#instagram_username").val() + " bot is running in this mode... Stop this mode and start again if you need change this configuration", buttons: [{ text: "OK" }] }).open();
        return 1;
    } else if (bot === "instagram" && check_err <= 0) {
        let tokens = instagram_get_user_form();
        let json = jtr.replace(tokens, require("../config.json"));
        // clean old logs
        fs.writeFile(remoteapp.getPath("userData") + "/" + $("#instagram_username").val() + "_" + $("#bot_mode").val() + ".log", "", function(err) {
            if (err) {
                return console.log(err);
            }
        });

        fs.writeFile(remoteapp.getPath("userData") + "/config_" + $("#instagram_username").val() + ".json", JSON.stringify(json), function(err) {
            if (err) {
                return console.log(err);
            }

            fs.exists($("#executable_path").val(), function(exists) {
                if (exists) {
                    app.dialog.create({ title: "Status", text: "Bot started...<br /><br />Example how check if work: wait 2min after start, open instagram app, go to settings, click on: Post you've liked. Alternative: click on LOGS button.<br /><br />Bot work for you :D", buttons: [{ text: "OK" }] }).open();
                    let id = tokens.instagram_username+"_"+tokens.bot_mode;
                    list_actived_bot[id] = true;
                    instagrambot_start(json, id);
                } else {
                    app.dialog.create({ title: "Warning", text: "Google Chrome path doesn't exist, please install google chrome or chromium and retry", buttons: [{ text: "OK" }] }).open();
                }
            });
        });
    }
}

function instagram_save_2fa() {

    fs.writeFile(remoteapp.getPath("userData") + "/" + $("#instagram_username").val() + "_pin.txt", $("#instagram_2fapin").val(), function(err) {
        if (err) {
            return console.log(err);
        }
    });
}

function instagram_open_logs() {
    clearInterval(logs_interval);
    $("#logs").val("[INFO] Loading...");
    logs_interval = setInterval(function() {
        let logs = fs.readFileSync(remoteapp.getPath("userData") + "/" + $("#instagram_username").val() + "_" + $("#bot_mode").val() + ".log", "utf8");
        $("#logs").val(logs);
    }, 1000);
}

function instagram_check_bot_mode() {
    if ($("#bot_mode").val() == "likemode_classic") {
        $(".bot_mode_desc").html("Bot go to random hashtag from list, like 1 photo and stop X minutes in loop");
        $(".likemode_all").hide();
        $("#bot_superlike_n").val(3);
        $("#bot_followday").val(300);
        $("#bot_followrotate").val(30);
        $("#bot_commentsday").val(300);
        $(".likemode_classic").show();
    } else if ($("#bot_mode").val() == "likemode_realistic") {
        $(".bot_mode_desc").html("Bot go to random hashtag from list, like 10-12 photo fast and stop X minutes in loop");
        $(".likemode_all").hide();
        $("#bot_superlike_n").val(3);
        $("#bot_followday").val(300);
        $("#bot_followrotate").val(30);
        $("#bot_commentsday").val(300);
        $(".likemode_realistic").show();
    } else if ($("#bot_mode").val() == "likemode_superlike") {
        $(".bot_mode_desc").html("Bot go to random hashtag from list, go to author profile, like 3 (configurable) random photos, return to hashtag list for 10-11 times and stop X minutes in loop");
        $(".likemode_all").hide();
        $("#bot_followday").val(300);
        $("#bot_followrotate").val(30);
        $("#bot_commentsday").val(300);
        $(".likemode_superlike").show();
    } else if ($("#bot_mode").val() == "likemode_competitor_users") {
        $(".bot_mode_desc").html("(Bot go to competitor account, like photos of his followers (10-12 photo fast and stop X minutes in loop))");
        $(".likemode_all").hide();
        $("#bot_superlike_n").val(3);
        $("#bot_followday").val(300);
        $("#bot_followrotate").val(30);
        $("#bot_commentsday").val(300);
        $(".likemode_competitor_users").show();
    } else if ($("#bot_mode").val() == "comment_mode") {
        $(".bot_mode_desc").html("Bot go to random hashtag from list, leave random comments from comments-list");
        $(".likemode_all").hide();
        $("#bot_superlike_n").val(3);
        $("#bot_followday").val(300);
        $("#bot_followrotate").val(30);
        $(".comment_mode").show();
    } else if ($("#bot_mode").val() == "fdfmode_classic") {
        $(".bot_mode_desc").html("Bot go to random hashtag from list, follow 30 users fast at 31 defollow first followed (number 1), follow 32, defollow number 2, in loop.");
        $(".likemode_all").hide();
        $("#bot_superlike_n").val(3);
        
        $(".fdfmode_classic").show();
    }
}

function instagram_check_hashtag() {
    return 1;
}

function instagram_check_superlike_n() {
    if ($("#bot_mode").val() == "likemode_superlike" && (parseInt($("#bot_superlike_n").val() * $("#bot_likeday_max").val()) >= 1000)) {
        app.dialog.create({ title: "Warning", text: $("#bot_superlike_n").val() + "x" + $("#bot_likeday_max").val() + "= is more than 1000 like/day. App restore default value.", buttons: [{ text: "OK" }] }).open();

        $("#bot_superlike_n").val(3);
        $("#bot_likeday_max").val(300);

        return 1;
    }

    return 0;
}

function instagram_check_max_like() {
    if ($("#bot_mode").val() == "likemode_superlike" && (parseInt($("#bot_superlike_n").val() * $("#bot_likeday_max").val()) >= 1000)) {
        app.dialog.create({ title: "Warning", text: $("#bot_superlike_n").val() + "x" + $("#bot_likeday_max").val() + "= is more than 1000 like/day. App restore default value.", buttons: [{ text: "OK" }] }).open();

        $("#bot_superlike_n").val(3);
        $("#bot_likeday_max").val(300);

        return 1;
    } else if ($("#bot_likeday_max").val() != "" && parseInt($("#bot_likeday_max").val()) > 1000) {
        $("#bot_likeday_max").val(900);

        app.dialog.create({ title: "Warning", text: "Instagram ban you if like more than 1000 photos/day. App restore default value.", buttons: [{ text: "OK" }] }).open();

        return 1;
    }

    return 0;
}


function instagram_check_likemode_competitor_users() {
    if ($("#bot_likemode_competitor_users").val() == "") {
        app.dialog.create({ title: "Warning", text: "Insert competitor username, restore default value", buttons: [{ text: "OK" }] }).open();

        $("#bot_likemode_competitor_users").val("ptkdev");

        return 1;
    }

    return 0;
}

function instagram_check_followrotate() {
    if ($("#bot_followrotate").val() > 3000) {
        app.dialog.create({ title: "Warning", text: "Value is overcapacity, restore default value", buttons: [{ text: "OK" }] }).open();

        $("#bot_followrotate").val(30);

        return 1;
    }

    return 0;
}

function instagram_check_followday() {
    if ($("#bot_followday").val() > 500) {
        app.dialog.create({ title: "Warning", text: "Value is overcapacity, restore default value", buttons: [{ text: "OK" }] }).open();

        $("#bot_followday").val(300);

        return 1;
    }

    return 0;
}

function instagram_check_commentsday() {
    if ($("#bot_commentsday").val() > 600) {
        app.dialog.create({ title: "Warning", text: "Value is overcapacity, restore default value", buttons: [{ text: "OK" }] }).open();

        $("#bot_commentsday").val(300);

        return 1;
    }

    return 0;
}

function instagram_check_userwhitelist() {
    return 1;
}


function instagram_check_comment_mode() {
    return 1;
}

